### ISSUE TITLE
Releasing 2.4.2

### ISSUE THREAD

**Issue by cvvergara** [pgRouting/pgrouting#907]
## prepare

- [x] [jenkins](http://winnie.postgis.net:1500/view/pgRouting/job/pgRouting_matrix_branch_2_4/)
- [x] make sure pull request are pulled
- [x] update release_notes and NEWS
- [x] make sure merges to release/2.4 are made
- [x] pump up version from 2.4.1 to 2.4.2 and push release/2.4
- [x] run the script

```
tools/release-scripts/check-release.sh 2 4 2 2.4.1 > ../release_results.txt
```
- [x] make pull request to master
- [x] accept pull request when appveyor, travis and jenkins are finished

## run the check script:

```
tools/release-scripts/check-release.sh 2 4 2 2.4.1 master > ../release_results.txt
```

- [x] write in a comment on this issue the results of the  check script

### make sure CI tests work

- [x] wait until all tests pass on travis, appveyor and jenkins and script finishes

### Documentation
- [x] verify locally correct versions
  - [x] users
  - [x] developers 
- [x] Copy the User's documentation to the gh-pages
```
git checkout gh-pages
rm -rf 2.4
mkdir 2.4
cp -r build/doc/html/en 2.4
git add 2.4
```
- [x]  commit users documentation

```
git commit --no-verify  -m  'Updating users documentation for 2.4.2'
```
- [x] Copy the Developer's documentation to the gh-pages
```
rm -Rf doxy/2.4
cp -r build/doxygen/html doxy/2.4
git add doxy/2.4
```
- [x]  commit developers documentation

```
git commit --no-verify  -m  'Updating developers documentation for 2.4.2'
```

- [x] latest documentation is 2.4.2

```
rm -rf  latest
mkdir latest
cp -r build/doc/html/en latest
git commit --no-verify  -a -m  'Updating the latest links with 2.4.2'
```

- [x] push & go to working branch

```
git push
git checkout master
```

- [x] verify [latest](http://docs.pgrouting.org/latest/en/index.html)
- [x] verify users [2.4](http://docs.pgrouting.org/2.4/en/index.html)
- [x] verify developers [2.4](http://docs.pgrouting.org/doxy/2.4/index.html)

### CI Tests
- [x]  [Travis](https://travis-ci.org/pgRouting/pgrouting/builds/264966769)
- [x]  [Jenkins](http://winnie.postgis.net:1500/view/pgRouting/job/pgRouting_matrix_master/200/)
- [x]  [appVeyor](https://ci.appveyor.com/project/Georepublic/pgrouting/build/2.4.956)

## THE TAG
- [x] create the tag & push it

```
git tag -a -m "Create v2.4.2 tag" v2.4.2
git push origin v2.4.2
```

- [x]  Edit the tag

https://github.com/pgRouting/pgrouting/releases/tag/v2.4.2

- [x] verify the links 
- [x] Modify the following changes if needed

```
Changes for 2.4.2
-------------------------------------------------------------------------------

*Improvement*

* Works for postgreSQL 10

*Bug fixes*

* Fixed: Unexpected error column "cname"
* Replace __linux__ with __GLIBC__ for glibc-specific headers and functions

```
## Announce 2.4.2
- [x] mail to  packagers
  - pgsql-pkg-yum
  - pgsql-pkg-debian
- [x] mail to:
  - pgrouting users
  - pgrouting dev
  - OSGeo Discuss
- [ ] tweet
- [ ] facebbook

## Close this issue
- [ ] Close this issue
- [ ] Close milestone 2.4.2

**Comment by cvvergara** [comment]
### Verify branch to be master

- [x] Working Branch: master

  - [x] Already in branch master

### Verify typos

```
sh tools/scripts/fix_typos.sh
```
- [x] Typos checked: OK

### Verify Current NEWS & CHANGLOG


- [x] Checking NEWS section 2.4.2 exists
```
grep 2.4.2 NEWS
```

- [x] Checking release_notes.rst file section 2.4.2 exists

```
grep 2.4.2 doc/src/release_notes.rst | grep ref
```


### Verify 2.4.1 NEWS & CHANGLOG

- [x] Checking NEWS section 2.4.1 exists

```
grep 2.4.1 NEWS
```

- [x] Checking release_notes.rst section 2.4.1 exists

```
* :ref:`changelog_2_4_1`
```


### Check version information

- [x] Check mayor information
```
cat CMakeLists.txt | grep 'set(PGROUTING_VERSION_MAJOR "2")'
```

- [x] Check minor information
```
cat CMakeLists.txt | grep 'set(PGROUTING_VERSION_MINOR "4")'
```

- [x] Check patch information
```
cat CMakeLists.txt | grep 'set(PGROUTING_VERSION_PATCH "2")'
```

```
cat CMakeLists.txt | grep 'set(PGROUTING_VERSION_DEV "")'
```
- [x] Check dev information


### Check pgr_version test result

```
cat doc/test/utilities-any.result | grep "2.4.2"
```
- [x] pgr_version result: OK

```
cat VERSION | grep "master"
```

  - [x] VERSION file branch: OK

### Checking signature files
- [x] tools/sigs/pgrouting--2.4.1.sig
- [x] tools/sigs/pgrouting--2.4.0.sig
- [x] tools/sigs/pgrouting--2.3.2.sig
- [x] tools/sigs/pgrouting--2.3.1.sig
- [x] tools/sigs/pgrouting--2.3.0.sig
- [x] tools/sigs/pgrouting--2.2.4.sig
- [x] tools/sigs/pgrouting--2.2.3.sig
- [x] tools/sigs/pgrouting--2.2.2.sig
- [x] tools/sigs/pgrouting--2.2.1.sig
- [x] tools/sigs/pgrouting--2.2.0.sig
- [x] tools/sigs/pgrouting--2.1.0.sig
- [x] tools/sigs/pgrouting--2.0.1.sig
- [x] tools/sigs/pgrouting--2.0.0.sig

### Locally make a clean build as Release

```
bash tools/release-scripts/compile-release.sh 5   2.4 2
bash tools/release-scripts/compile-release.sh 4.9 2.4 2
bash tools/release-scripts/compile-release.sh 4.6 2.4 2
bash tools/release-scripts/compile-release.sh 4.8 2.4 2
```


Compiling with 5
------------------------------------
update-alternatives: using /usr/bin/gcc-5 to provide /usr/bin/gcc (gcc) in manual mode
  - [x] Compilation OK
  - [x] Documentation tests OK
  - [x] Pgtap tests OK


Compiling with 4.9
------------------------------------
update-alternatives: using /usr/bin/gcc-4.9 to provide /usr/bin/gcc (gcc) in manual mode
  - [x] Compilation OK
  - [x] Documentation tests OK
  - [x] Pgtap tests OK

Compiling with 4.6
------------------------------------
update-alternatives: using /usr/bin/gcc-4.6 to provide /usr/bin/gcc (gcc) in manual mode
  - [x] Compilation OK
  - [x] Documentation tests OK
  - [x] Pgtap tests OK

Compiling with 4.8
------------------------------------
update-alternatives: using /usr/bin/gcc-4.8 to provide /usr/bin/gcc (gcc) in manual mode
  - [x] Compilation OK
  - [x] Documentation tests OK
  - [x] Pgtap tests OK
  - [x] Build Users documentation OK
  - [x] Build developers documentation OK
  - [x] completed local builds

### checking the signature files dont change

- [x] completed check: OK

### Locally run the update tester

```
bash tools/testers/update-tester.sh
```

- [ ] All pgRouting versions to be updated must be installed before calling

Updating from 2.4.1 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.4.1.sql found
CREATE EXTENSION
CREATE EXTENSION
              pgr_version               
----------------------------------------
 (2.4.1,v2.4.1,a1f6835e5,master,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```

Updating from 2.4.0 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.4.0.sql found
CREATE EXTENSION
CREATE EXTENSION
                 pgr_version                 
---------------------------------------------
 (2.4.0,v2.4.0,4e9518cd3,release/2.4,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```

Updating from 2.3.1 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.3.1.sql found
CREATE EXTENSION
CREATE EXTENSION
             pgr_version              
--------------------------------------
 (2.3.1,v2.3.1,963f95d,master,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```

Updating from 2.3.0 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.3.0.sql found
CREATE EXTENSION
CREATE EXTENSION
                  pgr_version                  
-----------------------------------------------
 (2.3.0,pgrouting-2.3.0,8c86efd,master,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```
Updating from 2.2.3 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.2.3.sql found
CREATE EXTENSION
CREATE EXTENSION
                  pgr_version                  
-----------------------------------------------
 (2.2.3,pgrouting-2.2.3,6d05b9b,master,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```

Updating from 2.2.2 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.2.2.sql found
CREATE EXTENSION
CREATE EXTENSION
                  pgr_version                  
-----------------------------------------------
 (2.2.2,pgrouting-2.2.2,544044b,master,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```

Updating from 2.2.1 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.2.1.sql found
CREATE EXTENSION
CREATE EXTENSION
                  pgr_version                  
-----------------------------------------------
 (2.2.1,pgrouting-2.2.1,7141904,master,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```

Updating from 2.2.0 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.2.0.sql found
CREATE EXTENSION
CREATE EXTENSION
                  pgr_version                  
-----------------------------------------------
 (2.2.0,pgrouting-2.2.0,afc622e,master,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```
Updating from 2.1.0 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.1.0.sql found
CREATE EXTENSION
CREATE EXTENSION
                   pgr_version                   
-------------------------------------------------
 (2.1.0,pgrouting-2.1.0,1,b38118a,master,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```

Updating from 2.0.0 to 2.4.2
------------------------------------
```
/usr/share/postgresql/9.3/extension/pgrouting--2.0.0.sql found
CREATE EXTENSION
CREATE EXTENSION
                   pgr_version                   
-------------------------------------------------
 (2.0.0,pgrouting-2.0.1,5,40ad51f,master,1.54.0)
(1 row)

ALTER EXTENSION
              pgr_version               
----------------------------------------
 (2.4.2,v2.4.2,fb7c27bfc,master,1.54.0)
(1 row)
```

Reached end of test, all tests passed
- [x] completed update testing
### execute the documentation queries
```
tools/testers/algorithm-tester.pl -documentation
git status
```
- [x] No files changed
End of check

**Comment by cvvergara** [comment]
@dersvenhesse
FYI: pgRouting 2.4.2 is released
PS. I don't have your e-mail.

**Comment by dersvenhesse** [comment]
@cvvergara, thanks. It's already merged in [homebrew-core#16865](https://github.com/Homebrew/homebrew-core/pull/16865). 🎉
